### ISSUE TITLE
descMetadata step must fail if missing required fields

### ISSUE THREAD

**Issue by ndushay** [sul-dlss/common-accessioning#372]
per uber ticket #188, the descMetadata step can complete "successfully" when it has partial (or entirely missing?) descriptiveMetadata created.

Example:

#### Incomplete descriptive metadata

From Ben:
> For an object like this: https://argo.stanford.edu/view/druid:bq454cj1549 - the catalog service call has produced MODS that is lacking all elements above the typeOfResource element and, therefore, does not produce a field that would go into DC - which is the source of the publish error.

![image](https://user-images.githubusercontent.com/96775/62987605-82c38680-bdf5-11e9-8c3d-da7fa3118426.png)

> Note that the user can’t fix this, because “Refresh descMetadata” is greyed out (because the object is in accessioning).

![image](https://user-images.githubusercontent.com/96775/62987592-72aba700-bdf5-11e9-95e6-aaa890c24c82.png)

Honeybadger:  https://app.honeybadger.io/projects/52894/faults/50003137

the corresponding dor-services-app  error: https://app.honeybadger.io/projects/50568/faults/49013058

Confirmed that this happens when I take the mods from `bq454cj1549`, which is 

```xml
      <mods xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.loc.gov/mods/v3" version="3.6" xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-6.xsd">
        <typeOfResource/>
        <originInfo>
          <issuance/>
        </originInfo>
        <recordInfo>
          <recordIdentifier source="SIRSI">a13132918</recordIdentifier>
          <recordOrigin>Converted from MARCXML to MODS version 3.6 using MARC21slim2MODS3-6_SDR.xsl (SUL version 1 2018/06/13; LC Revision 1.118 2018/01/31)</recordOrigin>
        </recordInfo>
      </mods>
```

and I run that mods xml through the dor-services-app DublinCoreService `.ng_xml` method, the error thrown is:

```
#<DublinCoreService::CrosswalkError: DublinCoreService#ng_xml produced incorrect xml (no children):
       <...ion="http://www.openarchives.org/OAI/2.0/oai_dc/ http://www.openarchives.org/OAI/2.0/oai_dc.xsd"/>
       > with backtrace:
         # ./app/services/dublin_core_service.rb:21:in `ng_xml'
         # ./spec/services/dublin_core_service_spec.rb:11:in `block (3 levels) in <top (required)>'
         # ./spec/services/dublin_core_service_spec.rb:49:in `block (4 levels) in <top (required)>'
         # ./spec/services/dublin_core_service_spec.rb:49:in `block (3 levels) in <top (required)>'
```

For this ticket:
- [ ] determine required descMetadata (MODS) fields required for subsequent steps to succeed
  - [ ] publishMetadata step (if object is to be published, only???)
  - [ ] rights, content, technical metadata?

**Comment by ndushay** [comment]
#### DC metadata
- cannot be empty 
- must have children
- generated from MODS and https://github.com/sul-dlss/dor-services-app/blob/master/app/services/mods2dc.xslt

- title:  mods `<titleInfo>` text contents must not be null (maybe `/titleInfo/title`, more specifically)

**Comment by ndushay** [comment]
Andrew, via slack:

> On descriptive metadata in general, this is the page I found that suggests that only title is truly required (at a system, if not project, level): https://consul.stanford.edu/display/DIGMETADATA/Replayable+spreadsheet+field-level+guidelines

**Comment by ndushay** [comment]
Merged PR #374 addresses the original intent of this ticket.

Closing this in favor of sul-dlss/dor-services-app#348 (raise error in dor-services-app if catalog lookup doesn't retrieve marc or retrieves incomplete marc) and #381 (require more than mods title if we have a catkey ... if we need it)
